config/main.py

```diff
@@ -1027,46 +1129,145 @@ def mirror_session():
 @click.argument('queue', metavar='[queue]', required=False)
 @click.option('--policer')
 def add(session_name, src_ip, dst_ip, dscp, ttl, gre_type, queue, policer):
-    """
-    Add mirror session
-    """
+    """ Add ERSPAN mirror session.(Legacy support) """
+    add_erspan(session_name, src_ip, dst_ip, dscp, ttl, gre_type, queue, policer)
+
+@mirror_session.group(cls=AbbreviationGroup, name='erspan')
+@click.pass_context
+def erspan(ctx):
+    """ ERSPAN mirror_session """
+    pass
+
+
+#
+# 'add' subcommand
+#
+
+@erspan.command('add')
+@click.argument('session_name', metavar='<session_name>', required=True)
+@click.argument('src_ip', metavar='<src_ip>', required=True)
+@click.argument('dst_ip', metavar='<dst_ip>', required=True)
+@click.argument('dscp', metavar='<dscp>', required=True)
+@click.argument('ttl', metavar='<ttl>', required=True)
+@click.argument('gre_type', metavar='[gre_type]', required=False)
+@click.argument('queue', metavar='[queue]', required=False)
+@click.argument('src_port', metavar='[src_port]', required=False)
+@click.argument('direction', metavar='[direction]', required=False)
+@click.option('--policer')
+def add(session_name, src_ip, dst_ip, dscp, ttl, gre_type, queue, policer, src_port, direction):
+    """ Add ERSPAN mirror session """
+    add_erspan(session_name, src_ip, dst_ip, dscp, ttl, gre_type, queue, policer, src_port, direction)
+
+def mirror_common_init(session_info, policer, queue, src_port, direction):
```

Comment:
I think `gather_session_info` might be a more descriptive name.

Reply:
Done.